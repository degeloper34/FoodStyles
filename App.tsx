import {ApolloClient, ApolloProvider, InMemoryCache} from "@apollo/client";
import {StatusBar} from "expo-status-bar";
import {SafeAreaProvider} from "react-native-safe-area-context";
import useCachedResources from "./src/hooks/useCachedResources";
import Navigation from "./src/navigation";

const token =
  "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NTI3LCJ0eXBlIjoiQUNDRVNTIiwiaWF0IjoxNjU3MTQ0MzAyLCJleHAiOjE2NTc3NDkxMDJ9.XxvR4ah2U6NeUFAxKtI-sSzByFfbHGcz06LGpp1FMPVCar6mQyfctQgtKOkg2w5Wv_luULa0SgcSovS6VNUB9_BBlFdH7PYBjQO3N1N0Dwi53FBzA0rdnSkRGdj5Oe7hQ-9fshljlo5cJUYBbz4N0Ndpt323V5Eu8lGqJZ8Z5husoCiK93fZ0l3-Vr0M_sLvulZAELXGRG1QmYCwMwrp43HF9zyOLjcSHPVCvbpettqGxGv8Gne7O-XbykNv8ehG81vPcJDMSImwXGY8Yd47EoIrxdoN-3IlJumBK1Ql29hy5N7HBbk2oil3KMwBbw4MowB7rdlaT1i5_5Xr6q0iIg";

export const client = new ApolloClient({
  uri: "https://api-dev.foodstyles.com/graphql",
  cache: new InMemoryCache(),
  headers: {
    authorization: `Bearer ${token}`,
  },
});

export default function App() {
  const isLoadingComplete = useCachedResources();

  if (!isLoadingComplete) {
    return null;
  } else {
    return (
      <ApolloProvider client={client}>
        <SafeAreaProvider>
          <StatusBar style={"light"} />
          <Navigation />
        </SafeAreaProvider>
      </ApolloProvider>
    );
  }
}
