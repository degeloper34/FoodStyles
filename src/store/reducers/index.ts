import {combineReducers} from "redux";

import memberReducers from "./memberReducers";

export const rootReducer = combineReducers({
  memberReducers,
});
