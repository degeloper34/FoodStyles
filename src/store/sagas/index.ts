import {all} from "redux-saga/effects";

import {memberWatcher} from "./memberSagas";

export default function* rootSaga() {
  yield all([memberWatcher()]);
}
