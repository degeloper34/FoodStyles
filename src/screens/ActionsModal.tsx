import {LinearGradient} from "expo-linear-gradient";
import {Alert, Image, Pressable, StyleSheet, View, Share} from "react-native";
import {CustomText} from "../components/atoms";
import {TextStyles} from "../styles";
import {useMutation} from "@apollo/client";
import {
  DELETE_CARD_MUTATION,
  DUPLICATE_CARD_MUTATION,
  SHARE_CARD_MUTATION,
} from "../api/mutations";
import {RootStackScreenProps} from "../../types";
import {client} from "../../App";
import {GET_CARDS_QUERY} from "../api/queries";
import {BlurView} from "expo-blur";
import {Ionicons} from "@expo/vector-icons";

export default function ActionsModal({
  navigation,
  route,
}: RootStackScreenProps<"ActionsModal">) {
  const {item} = route.params;

  const [share] = useMutation(SHARE_CARD_MUTATION);
  const [duplicate] = useMutation(DUPLICATE_CARD_MUTATION);
  const [remove] = useMutation(DELETE_CARD_MUTATION);

  const onPressGoBack = () => {
    navigation.goBack();
  };

  const onPressShare = async () => {
    await share({variables: {id: item.id}});

    await Share.share({
      message: item?.name,
    });
  };

  const onPressDuplicate = async () => {
    await duplicate({variables: {id: item.id}});
    await reFetchCards();

    navigation.pop();
  };

  const onPressRemove = async () => {
    Alert.alert(
      "Confirm delete",
      "This will delete the Food Style and all its settings.",
      [
        {
          text: "Delete",
          onPress: onPressDelete,
          style: "destructive",
        },
        {text: "Cancel", onPress: () => console.log("Cancel Pressed")},
      ]
    );
  };

  const onPressDelete = async () => {
    await remove({variables: {id: item.id}});
    await reFetchCards();

    navigation.pop();
  };

  const reFetchCards = async () => {
    await client.refetchQueries({
      include: [GET_CARDS_QUERY],
    });
  };

  const {
    container,
    btnBlur,
    viewBlur,
    linearGradient,
    txtName,
    viewCardName,
    viewActionsContainer,
    viewAction,
  } = styles;

  return (
    <View style={container}>
      <Pressable style={btnBlur} onPress={onPressGoBack}>
        <BlurView intensity={20} style={viewBlur} />
      </Pressable>

      <LinearGradient
        colors={["rgb(250,119,69)", "rgb(243, 196, 66)"]}
        style={linearGradient}
      >
        <View style={viewCardName}>
          <CustomText
            text={item.name}
            type={"semi-bold"}
            style={[txtName, TextStyles.TEXT_STYLE_6]}
            numberOfLines={2}
          />
        </View>

        <View style={viewActionsContainer}>
          <View style={viewAction}>
            <Pressable onPress={onPressGoBack}>
              <Ionicons
                name="chevron-back-circle-sharp"
                size={50}
                color="white"
              />
            </Pressable>

            <CustomText
              text={"Back"}
              type={"semi-bold"}
              style={TextStyles.TEXT_STYLE_6}
            />
          </View>

          <View style={viewAction}>
            <Pressable onPress={onPressShare}>
              <Image source={require("../assets/images/share.png")} />
            </Pressable>

            <CustomText
              text={"Share"}
              type={"semi-bold"}
              style={TextStyles.TEXT_STYLE_6}
            />
          </View>

          <View style={viewAction}>
            <Pressable onPress={onPressDuplicate}>
              <Image source={require("../assets/images/duplicate.png")} />
            </Pressable>

            <CustomText
              text={"Duplicate"}
              type={"semi-bold"}
              style={TextStyles.TEXT_STYLE_6}
            />
          </View>

          <View style={viewAction}>
            <Pressable onPress={onPressRemove}>
              <Image source={require("../assets/images/delete.png")} />
            </Pressable>

            <CustomText
              text={"Delete"}
              type={"semi-bold"}
              style={TextStyles.TEXT_STYLE_6}
            />
          </View>
        </View>
      </LinearGradient>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  btnBlur: {
    flex: 3,
  },
  viewBlur: {
    flex: 1,
    backgroundColor: "transparent",
  },
  linearGradient: {
    flex: 1,
  },
  viewCardName: {
    alignItems: "center",
    padding: 16,
    justifyContent: "center",
  },
  txtName: {
    textAlign: "center",
  },
  viewActionsContainer: {
    flexDirection: "row",
  },
  viewAction: {
    flex: 1,
    alignItems: "center",
  },
});
