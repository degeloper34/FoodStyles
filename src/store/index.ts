export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

import {legacy_createStore as createStore} from "redux";
import {rootReducer} from "./reducers";

export const store = createStore(rootReducer);
