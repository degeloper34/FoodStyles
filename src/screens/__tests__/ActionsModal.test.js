import renderer from "react-test-renderer";
import React from "react";
import {Provider} from "react-redux";
import {store} from "../../store";
import ActionsModal from "../ActionsModal";
import {ApolloProvider} from "@apollo/client";
import {apolloClient} from "../../../App";

describe("ActionsModal Tests", () => {
  test("HomeScreen | Render without issues", () => {
    const screen = renderer
      .create(
        <ApolloProvider client={apolloClient}>
          <Provider store={store}>
            <ActionsModal
              route={{
                params: {
                  item: {
                    id: 1,
                    name: "Test",
                  },
                },
              }}
            />
          </Provider>
        </ApolloProvider>
      )
      .toJSON();

    expect(screen).toMatchSnapshot();
  });
});
