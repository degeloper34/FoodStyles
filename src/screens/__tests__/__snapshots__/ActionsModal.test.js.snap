// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionsModal Tests HomeScreen | Render without issues 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
          },
        ]
      }
    >
      <ViewManagerAdapter_ExpoBlurView
        proxiedProperties={
          Object {
            "intensity": 20,
            "tint": "default",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4294604613,
          4294165570,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "padding": 16,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            Array [
              Object {
                "textAlign": "center",
              },
              Object {
                "color": "#434343",
                "fontSize": 18,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 22,
              },
            ],
            Object {
              "fontFamily": "proxima-semibold",
              "textDecorationLine": "none",
            },
          ]
        }
      >
        Test
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        />
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#434343",
                "fontSize": 18,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 22,
              },
              Object {
                "fontFamily": "proxima-semibold",
                "textDecorationLine": "none",
              },
            ]
          }
        >
          Back
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Image
            source={1}
          />
        </View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#434343",
                "fontSize": 18,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 22,
              },
              Object {
                "fontFamily": "proxima-semibold",
                "textDecorationLine": "none",
              },
            ]
          }
        >
          Share
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Image
            source={1}
          />
        </View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#434343",
                "fontSize": 18,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 22,
              },
              Object {
                "fontFamily": "proxima-semibold",
                "textDecorationLine": "none",
              },
            ]
          }
        >
          Duplicate
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Image
            source={1}
          />
        </View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#434343",
                "fontSize": 18,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 22,
              },
              Object {
                "fontFamily": "proxima-semibold",
                "textDecorationLine": "none",
              },
            ]
          }
        >
          Delete
        </Text>
      </View>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
